containerized_ml_models/model_api_2/api.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask, jsonify, request
+from pickle import load as pkl_load
+from ast import literal_eval
+from numpy import asarray
+
+app = Flask(__name__)
+MODEL_NAME = "model.pkl"
+
+
+def load_pickle(fileName):
+    return pkl_load(open(MODEL_NAME, "rb"))
```

Comment:
Funkcja z argumentem, którego nie używasz tylko argumentgu globlnego, a w call'u funkcji w predict() nie ma argumentu brakującego?

Reply:
Przy okazji nie jestem pewny jak dla tego przypadku zadziała garbage collector? Zazwyczaj otwierajac plik stosuje sie klauzule with open(filename) as file:, bo wprost mówimy kiedy ma python zamknąć plik, a tu potencjalnie będzie wisiał aż śmieciara przyjedzie. No chyba że w ten sposób pisząc samo się to zamyka?

Reply:
> Funkcja z argumentem, którego nie używasz tylko argumentgu globlnego, a w call'u funkcji w predict() nie ma argumentu brakującego?

Faktycznie, słuszna uwaga. Czy w takim razie, dobrą praktyką będzie zastosowanie argumentu globalnego, czy podanie nazwy pliku (nie przewidujemy raczej, że będzie on miał inną nazwę, a wykorzystujemy go raz) jako argumentu funkcji, będzie poprawnym podejściem?

Reply:
> Przy okazji nie jestem pewny jak dla tego przypadku zadziała garbage collector? Zazwyczaj otwierajac plik stosuje sie klauzule with open(filename) as file:, bo wprost mówimy kiedy ma python zamknąć plik, a tu potencjalnie będzie wisiał aż śmieciara przyjedzie. No chyba że w ten sposób pisząc samo się to zamyka?

I tak też powinno być tutaj, masz rację, tutaj gc dopiero posprząta po mnie. Dzięki za uwagę.

Reply:
Ja po prostu zhardkodowałem to w 'with open', lepiej na pewno jakas zmienną do tego mieć. Czy globalna czy argument do funkcji to kwestia jak dalej rozwijane byłoby wykorzystanie tego pliku, gdzie potrzebna nazwa itd. itp. Teraz chyba dość obojętny jest wybór